src/gui/menu_color.cpp

```diff
@@ -14,30 +15,108 @@
 //  You should have received a copy of the GNU General Public License
 //  along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
+#include <sstream>
+
 #include "gui/menu_color.hpp"
+#include "gui/menu_item.hpp"
+#include "util/log.hpp"
 
 #include "util/gettext.hpp"
 
-ColorMenu::ColorMenu(Color* color_) :
-  color(color_)
+ColorMenu::ColorMenu(Color* color) :
+  m_color(color)
 {
   add_label(_("Mix the colour"));
   add_hl();
 
-  add_color_picker_2d(*color);
-  add_color_channel_rgba(&(color->red), Color::RED);
-  add_color_channel_rgba(&(color->green), Color::GREEN);
-  add_color_channel_rgba(&(color->blue), Color::BLUE);
-  add_color_channel_rgba(&(color->alpha), Color::BLACK, -1, true);
-  add_color_display(color);
+  add_color_picker_2d(*m_color);
+  add_color_channel_rgba(&(m_color->red), Color::RED);
+  add_color_channel_rgba(&(m_color->green), Color::GREEN);
+  add_color_channel_rgba(&(m_color->blue), Color::BLUE);
+  add_color_channel_rgba(&(m_color->alpha), Color::BLACK, -1, true);
+  add_color_display(m_color);
+
+  add_hl();
+  add_entry(MNID_COPY_CLIPBOARD_RGB, _("Copy to clipboard (rgb)"));
+  add_entry(MNID_COPY_CLIPBOARD_HEX, _("Copy to clipboard (hex)"));
+  if (SDL_HasClipboardText())
+  {
+    char* clipboard_text = SDL_GetClipboardText();
+    std::optional<Color> clipboard_color;
+
+    if (clipboard_text)
+    {
+      const std::string text(clipboard_text);
+      SDL_free(clipboard_text);
+
+      clipboard_color = Color::deserialize_from_rgb(text);
+      if (!clipboard_color)
+        clipboard_color = Color::deserialize_from_hex(text);
+    }
+
+    add_entry(MNID_PASTE_CLIPBOARD, _("Paste from clipboard"), clipboard_color.value_or(Color(1.f, 1.f, 1.f)));
+  }
+  else
+    add_entry(MNID_PASTE_CLIPBOARD, _("Paste from clipboard"), Color(1.f, 1.f, 1.f));
```

Comment:
What's the point to add an item called "Paste from clipboard" here when SDL doesn't support the functionality?

Reply:
SDL_GetClipboardText(). What do you mean???

Reply:
Never mind. I misunderstood the meaning of `SDL_HasClipboardText()`.